bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/XMLNonElementItem.java

```diff
@@ -0,0 +1,241 @@
+import org.ballerinalang.jvm.values.api.BXML;
+import org.ballerinalang.jvm.values.freeze.Status;
+
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import static org.ballerinalang.jvm.util.BLangConstants.STRING_NULL_VALUE;
+
+/**
+ * Functionality common to PI, COMMENT and TEXT nodes.
+ *
+ * @since 1.2.0
+ */
+public abstract class XMLNonElementItem extends XMLValue {
+    @Override
+    public abstract boolean isEmpty();
+
+    @Override
+    public boolean isSingleton() {
+        return false;
+    }
+
+    @Override
+    public String getItemType() {
+        return getNodeType().value();
+    }
+
+    @Override
+    public String getElementName() {
+        return "";
+    }
+
+    @Override
+    public abstract String getTextValue();
+
+    @Override
+    public String getAttribute(String localName, String namespace) {
+        return null;
+    }
+
+    @Override
+    public String getAttribute(String localName, String namespace, String prefix) {
+        return null;
+    }
+
+    @Override
+    public void setAttribute(String localName, String namespace, String prefix, String value) {
+
+    }
+
+    @Override
+    public MapValue<String, String> getAttributesMap() {
+        return null;
+    }
+
+    @Override
+    public void setAttributes(BMap<String, ?> attributes) {
+
+    }
+
+    @Override
+    public BXML elements() {
+        return new XMLSequence();
+    }
+
+    @Override
+    public BXML elements(String qname) {
+        return null;
+    }
+
+    @Override
+    public XMLValue children() {
+        return new XMLSequence();
+    }
+
+    @Override
+    public XMLValue children(String qname) {
+        return new XMLSequence();
+    }
+
+    @Override
+    public void setChildren(BXML seq) {
+    }
+
+    @Override
+    @Deprecated
+    public void addChildren(BXML seq) {
+    }
+
+    @Override
+    public BXML strip() {
+        return new XMLText("");
+    }
+
+    @Override
+    public abstract XMLNodeType getNodeType();
+
+    @Override
+    public BXML slice(long startIndex, long endIndex) {
+        return null;
+    }
+
+    @Override
+    public BXML descendants(String qname) {
+        return null;
+    }
+
+    @Override
+    public XMLValue getItem(int index) {
+        return null;
+    }
+
+    @Override
+    public Object copy(Map<Object, Object> refs) {
+        // XMLContentHolderItem is immutable
+        return this;
+    }
+
+    @Override
+    public Object frozenCopy(Map<Object, Object> refs) {
+        return this;
+    }
+
+    @Override
+    public int size() {
+        return 1;
+    }
+
+    @Override
+    public void build() {
+
+    }
+
+    @Override
+    public void removeAttribute(String qname) {
+
+    }
+
+    @Override
+    @Deprecated
+    public void removeChildren(String qname) {
+
+    }
+
+    @Override
+    public abstract OMNode value();
```

Comment:
Do we need this?

Reply:
This is still being used for testing (Jvm values to BVM values conversion)